### PR TITLE
Add ColorBar JS API

### LINKED ISSUE #4989: Add ColorBar public JS API definition
Add ColorBar public JS API definition

### PR DESCRIPTION
**Pull request by canavandl**
Add ColorBar JS API

fixes #4989

### CODE REVIEW

File: bokehjs/src/coffee/api/typings/models/annotations.d.ts [hunk 1]
```diff
@@ -54,6 +54,96 @@ declare namespace Bokeh {
...
     }
 
+    export var ColorBar: { new(attributes?: IColorBar, options?: ModelOpts): ColorBar };
+    export interface ColorBar extends Annotation, IColorBar {}
+    export interface IColorBar extends IAnnotation {
+        location?: LegendLocation;
+        orientation?: Orientation;
+
+        legend_height?: Int;
+        legend_width?: Int;
```
Comment (mattpap) [inline bokehjs/src/coffee/api/typings/models/annotations.d.ts]: Why color bar has `legend_*` properties? I thought this was some WIP stuff in the original PR.

**Comment by mattpap** [comment]
LGTM

File: bokehjs/src/coffee/api/typings/models/annotations.d.ts (hunk 1 above)
Comment (bryevdv) [inline bokehjs/src/coffee/api/typings/models/annotations.d.ts]: Is this intentional, or copypasta?

Comment (canavandl) [inline bokehjs/src/coffee/api/typings/models/annotations.d.ts]: It was intentional, but not a good idea. I've since removed the legend_* prefixes

**Comment by birdsarah** [comment]
+1 on the legend_prefix removal - is inline with this proposal: https://github.com/bokeh/bokeh/issues/4526

**Pull request merged**